drivers/clock_control/clock_control_tt_bh.c

```diff
@@ -0,0 +1,624 @@
+/*
+ * Copyright (c) 2025 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT tenstorrent_bh_clock_control
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock-control/clock_control_tt_bh.h>
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/sys/util.h>
+#include <stdint.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clock_control_tt_bh);
+
+#define PLL_COUNT 5
+
+#define PLL_LOCK_TIMEOUT 400
```

Comment:
Would be good to add units to this
```suggestion
#define PLL_LOCK_TIMEOUT_MS 400
```